### PR TITLE
Create sphinx documentation

### PR DESCRIPTION
**Pull request by DavidKnott**
The goal of this PR is to facilitate contributions to the docs by creating a base starting point (using `sphinx` and `recommonmark` ).
To runs the docs locally enter:
1) `pip install -r requirements-docs.txt`
2) `make docs`
@yograterol

### CODE REVIEW

File: docs/conf.py [hunk 1]
```diff
@@ -0,0 +1,180 @@
...
+# If your documentation needs a minimal Sphinx version, state it here.
+#
+# needs_sphinx = '1.0'
+
+# Add any Sphinx extension module names here, as strings. They can be
+# extensions coming with Sphinx (named 'sphinx.ext.*') or your custom
+# ones.
+extensions = [
+    'sphinx.ext.autodoc'
+    ]
```
Comment (yograterol) [inline docs/conf.py]: PEP8, 
```
extensions = [
    'sphinx.ext.autodoc'
]
```

**Comment by yograterol** [comment]
@DavidKnott I see this good, just solve the conflicts. We could create an issue with the points to document and hand out the tasks.

**Comment by yograterol** [comment]
image

:+1:

**Comment by DavidKnott** [comment]
@yograterol I fixed the conflicts and the styling.  Good thinking, let's split up the doc work into manageable issues.

**Pull request merged**